tests/functional/aws-node-sdk/test/object/post.js

```diff
@@ -0,0 +1,1621 @@
+        { name: 'x-amz-signature', value: signature },
+        { name: 'x-amz-date', value: formattedDate },
+        { name: 'policy', value: policyBase64 },
+        { name: 'bucket', value: bucket },
+        { name: 'key', value: key },
+    ];
+    if (!additionalConditions) {
+        return returnFields;
+    }
+    if (additionalConditions) {
+        additionalConditions.forEach(field => {
+            const key = Object.keys(field)[0];
+            const value = field[key];
+            const index = returnFields.findIndex(f => f.name === key);
+            if (index !== -1) {
+                returnFields[index].value = value;
+            } else {
+                returnFields.push({ name: key, value });
+            }
+        });
+    }
+    return returnFields;
+};
+
+describe('POST object', () => {
+    let bucketUtil;
+    let config;
+    const testContext = {};
+
+    before(() => {
+        config = getConfig('default');
+        ak = config.credentials.accessKeyId;
+        sk = config.credentials.secretAccessKey;
+        bucketUtil = new BucketUtility('default');
+        s3 = bucketUtil.s3;
+    });
+
+    beforeEach(done => {
+        bucketName = generateBucketName();
+        const url = `${config.endpoint}/${bucketName}`;
+        testContext.bucketName = bucketName;
+        testContext.url = new URL(url);
+
+        const fileContent = 'This is a test file';
+        fileBuffer = Buffer.from(fileContent);
+
+        // Create the bucket
+        s3.createBucket({ Bucket: bucketName }, err => {
+            if (err) {
+                return done(err);
+            }
+            return done();
+        });
+    });
+
+
+    afterEach(done => {
+        const { bucketName } = testContext;
+
+        process.stdout.write('Emptying bucket\n');
+        bucketUtil.empty(bucketName)
+            .then(() => {
+                process.stdout.write('Deleting bucket\n');
+                return bucketUtil.deleteOne(bucketName);
+            })
+            .then(() => done())
+            .catch(err => {
+                if (err.code !== 'NoSuchBucket') {
+                    process.stdout.write('Error in afterEach\n');
+                    return done(err);
+                }
+                return done();
+            });
+    });
+
+    it('should successfully upload an object using a POST form', done => {
+        const { url } = testContext;
+        const fields = calculateFields(ak, sk);
+        const formData = new FormData();
+
+        fields.forEach(field => {
+            formData.append(field.name, field.value);
+        });
+
+        formData.append('file', fileBuffer, { filename });
+
+        formData.getLength((err, length) => {
+            if (err) {
+                return done(err);
+            }
+
+            const options = {
+                method: 'POST',
+                hostname: url.hostname,
+                port: url.port,
+                path: url.pathname + url.search,
```

Comment:
Review the documentation: https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html#RESTObjectPOST-requests

> ```
> POST / HTTP/1.1
> Host: destinationBucket.s3.amazonaws.com
> ```

Anything (more than `/`) in the path should return a 405 (see https://github.com/scality/cloudserver/pull/5601#discussion_r1683325788 and https://github.com/scality/Arsenal/pull/2248#discussion_r1681347282)

And any querystring (`url.search`) should return a 400 (see https://github.com/scality/Arsenal/pull/2248#discussion_r1670771248)

Don't put them in tests if you don't want them to fail once url validation is correct